src/assessments/seo/internalLinksAssessment.js

```diff
@@ -1,99 +1,132 @@
-var AssessmentResult = require( "../../values/AssessmentResult.js" );
-var isEmpty = require( "lodash/isEmpty" );
+const Assessment = require( "../../assessment.js" );
+const AssessmentResult = require( "../../values/AssessmentResult.js" );
+const merge = require( "lodash/merge" );
 
 /**
- * Returns a score and text based on the linkStatistics object.
- *
- * @param {object} linkStatistics The object with all linkstatistics.
- * @param {object} i18n The object used for translations
- * @returns {object} resultObject with score and text
+ * Assessment to check whether the text has internal links and whether they are followed or no-followed.
  */
-var calculateLinkStatisticsResult = function( linkStatistics, i18n ) {
-	const url = "<a href='https://yoa.st/2pm' target='_blank'>";
+class TextHasInternalLinksAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
 
-	if ( linkStatistics.internalTotal === 0 ) {
-		return {
-			score: 3,
-			text: i18n.sprintf(
-				/* Translators: %1$s expands to a link on yoast.com, %2$s expands to the anchor end tag */
-				i18n.dgettext( "js-text-analysis", "No %1$sinternal links%2$s appear in this page, consider adding some as appropriate." ),
-				url,
-				"</a>"
-			),
+		let defaultConfig = {
+			parameters: {
+				recommendedMinimum: 1,
+			},
+			scores: {
+				allInternalFollow: 9,
+				someInternalFollow: 8,
+				noneInternalFollow: 7,
+				noInternal: 3,
+			},
+			url: "<a href='https://yoa.st/2pm' target='_blank'>",
 		};
+
+		this.identifier = "internalLinks";
+		this._config = merge( defaultConfig, config );
 	}
 
-	if ( linkStatistics.internalNofollow === linkStatistics.total ) {
-		return {
-			score: 7,
-			text: i18n.sprintf(
-				/* Translators: %1$s expands the number of internal links, %2$s expands to a link on yoast.com, %3$s expands to the anchor end tag */
-				i18n.dgettext( "js-text-analysis", "This page has %1$s %2$sinternal link(s)%3$s, all nofollowed." ),
-				linkStatistics.internalNofollow,
-				url,
-				"</a>"
-			),
-		};
+	/**
+	 * Runs the getLinkStatistics module, based on this returns an assessment result with score.
+	 *
+	 * @param {Paper} paper The paper to use for the assessment.
+	 * @param {Researcher} researcher The researcher used for calling research.
+	 * @param {Object} i18n The object used for translations.
+	 *
+	 * @returns {Object} The AssessmentResult.
+	 */
+	getResult( paper, researcher, i18n ) {
```

Comment:
I reason it's because the Assessment is called through calling its getResult
`getResult( paper, researcher, i18n )`. If you have a clear idea how Assessment.js and the files that call it should be changed to make things more efficient, let's do it in a separate PR.